sdk/azcore/to/to.go

```diff
@@ -0,0 +1,107 @@
+//go:build go1.16
+// +build go1.16
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
```

Comment:
Add a comment into this file that this package will be deprecated once Go adds generics

Reply:
The package won't be deprecated, just all the per-type functions.  We'll end up with one generic `To[T any](t T) *T` function.